ci/build_container/build_container_ubuntu.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+set -e
+
+# Setup basic requirements and install them.
+apt-get update
+apt-get install -y wget software-properties-common make cmake git python python-pip \
+  bc libtool automake zip time golang g++ gdb strace
+# clang head (currently 5.0)
+wget -O - http://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -
+apt-add-repository "deb http://apt.llvm.org/xenial/ llvm-toolchain-xenial-5.0 main"
+apt-get update
+apt-get install -y clang-5.0 clang-format-5.0
+# Bazel and related dependencies.
+apt-get install -y openjdk-8-jdk curl
+echo "deb [arch=amd64] http://storage.googleapis.com/bazel-apt stable jdk1.8" | tee /etc/apt/sources.list.d/bazel.list
+curl https://bazel.build/bazel-release.pub.gpg | apt-key add -
+apt-get update
+apt-get install -y bazel
+rm -rf /var/lib/apt/lists/*
+
+# virtualenv
+pip install virtualenv
+
+EXPECTED_CXX_VERSION="g++ (Ubuntu 5.4.0-6ubuntu1~16.04.4) 5.4.0 20160609" ./build_container_common.sh
```

Comment:
Is this really true? Aren't we using Clang in this container?

Reply:
the *recipes* are bulit using g++ from https://github.com/envoyproxy/envoy/blob/dc606268c87b259da1ef277bbbb1d9235b7f578f/ci/build_container/build_container.sh#L29

i keep this behaviour for both ubuntu/centos images.

i guess the clang is only used for bazel.{asan,tsan,dev}/{fix,check}_format target,
i made the necessary symbolic links in the centos image for this.